lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -185,28 +185,31 @@ def package_manager
       end
 
       def os_facts
-        facter = ForemanMaintain::Utils::Facter.path
-        @os_facts ||= JSON.parse(execute("#{facter} -j os"))
+        @os_facts ||= ForemanMaintain::Utils::OsFacts
       end
 
       def el?
-        os_facts['os']['family'] == 'RedHat'
+        ids = ['rhel', 'rhel fedora']
+        ids.include?(os_facts.id_like) ||
+          ids.include?(os_facts.id)
       end
 
       def debian?
-        os_facts['os']['family'] == 'Debian'
+        ids = %w[debian ubuntu]
+        ids.include?(os_facts.id_like) ||
+          ids.include?(os_facts.id)
       end
 
       def el7?
-        os_facts['os']['release']['major'] == '7' && el?
+        os_facts.version_id.to_i == 7 && el?
```

Comment:
Why not implement this as 
```suggestion
        el_major_version == 7
```